src/metaopt/io/database/mongodb.py

```diff
@@ -188,4 +188,8 @@ def remove(self, collection_name, query):
         Returns:
         result -- the document/documents
         """
-        self.db[collection_name].remove(query)
+        result = self.db[collection_name].find_one_and_delete(filter=query,
```

Comment:
To complement this [comment](https://github.com/mila-udem/metaopt/pull/8#discussion_r152851453), I feel like deletion are very likely to be requested on many documents and we are getting our hands tied here.

Reply:
Idem. But I'd rather return the document first at the end of the function rather than using `delete_one` or `delete_many`.

Reply:
Being able to recover the deleted document could be handy but I'm not sure we are likely to encounter such a use case.

Reply:
Yes, although rarely. I'm seeing a case where we want to simply remove a `node` from the set of `resources` if suddenly it becomes idle, we need to do a remove from the `resources`, but then if the `node` shows up again on our ping, we might need to return it back again. Just removing it might be dangerous.

Reply:
I don't know. If ressources are represented as objects which can easily register themselves in the database then it would do no harm to remove it and register again using the objects info rather than the deleted rows info. To me the best point in favor of find_and_remove is if we want to delete a row and make sure nobody changes it meanwhile. If we do db.load() and then db.remove() it could have changed in between. But again, how is that likely to happen given our use cases and is it worth limiting the flexibility of our interface? 

@dendisuhubdy @tsirif What would you think about setting base methods (insert, load, update, remove) as non-atomic ones supporting many operations and latter on we extend the interface to find_and_(update, remove) if needed?